src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs

```diff
@@ -121,7 +128,21 @@ protected EthereumTestResult RunTest(GeneralStateTest test, ITxTracer txTracer)
                 };
                 header.ExcessBlobGas = BlobGasCalculator.CalculateExcessBlobGas(parent, spec);
             }
-            bool isValid = txValidator.IsWellFormed(test.Transaction, spec);
+
+            Block block = Build.A.Block.WithTransactions(test.Transaction).WithHeader(header).TestObject;
+            IBlockTree blockTree = Build.A.BlockTree()
+                .WithSpecProvider(specProvider)
+                .WithoutSettingHead
+                .TestObject;
+
+            var difficultyCalculator = new EthashDifficultyCalculator(specProvider);
+            var sealer = new EthashSealValidator(_logManager, difficultyCalculator, new CryptoRandom(), new Ethash(_logManager), Timestamper.Default);
+            IHeaderValidator headerValidator = new HeaderValidator(blockTree, sealer, specProvider, _logManager);
+            IUnclesValidator unclesValidator = new UnclesValidator(blockTree, headerValidator, _logManager);
+            IBlockValidator blockValidator = new BlockValidator(txValidator, headerValidator, unclesValidator, specProvider, _logManager);
+
+            bool isValid = txValidator.IsWellFormed(test.Transaction, spec) && blockValidator.ValidateOrphanedBlock(block, out _);
```

Comment:
Can you hide new parts to some ValidateBlock method?

Reply:
refactored